kernel/objects/capability-spinning/objects/CapEntry.hh

```diff
@@ -185,13 +259,18 @@ namespace mythos {
                                        CapEntry& targetEntry, Cap targetCap, 
                                        COMMITFUN const& commit)
   {
+    MLOG_ERROR(mlog::cap, __PRETTY_FUNCTION__, DVAR(this), DVAR(parentCap), DVAR(targetEntry), DVAR(targetCap));
     ASSERT(isKernelAddress(this));
     ASSERT(targetEntry.cap().isAllocated());
-    lock(); // lock the parent entry, the child is already acquired
+    // lock the parent entry, the child is already acquired
+    lock_cap();
+    lock_next(); 
```

Comment:
> can retry infinitely without deadlocking because...

What do you mean with that?

We might live-lock if we keep trying. However: if we lost the race, that means the parent is either dead or moved somewhere else. No need to retry, the user has to handle this.

Reply:
`lock_cap()` is called here, which is an unbounded loop until `try_lock_cap()` is successfull. How would this identify the lost race?